discussions.go

```diff
@@ -0,0 +1,477 @@
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/issues/%d/discussions/%d", url.QueryEscape(project), issue, discussion)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	n := new(Discussion)
+	resp, err := s.client.Do(req, n)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return n, resp, err
+}
+
+// CreateIssueDiscussionOptions represents the available CreateIssueDiscussion()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#create-new-issue-discussion
+type CreateIssueDiscussionOptions struct {
+	Body *string `url:"body,omitempty" json:"body,omitempty"`
+}
+
+// CreateIssueDiscussion creates a new discussion to a single project issue.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#create-new-issue-discussion
+func (s *DiscussionsService) CreateIssueDiscussion(pid interface{}, issue int, opt *CreateIssueDiscussionOptions, options ...OptionFunc) (*Discussion, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/issues/%d/discussions", url.QueryEscape(project), issue)
+
+	req, err := s.client.NewRequest("POST", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	n := new(Discussion)
+	resp, err := s.client.Do(req, n)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return n, resp, err
+}
+
+// UpdateIssueDiscussionOptions represents the available UpdateIssueDiscussion()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#modify-existing-issue-discussion
+type UpdateIssueDiscussionOptions struct {
+	Body *string `url:"body,omitempty" json:"body,omitempty"`
+}
+
+// UpdateIssueDiscussion modifies existing discussion of an issue.
+//
+// https://docs.gitlab.com/ce/api/discussions.html#modify-existing-issue-discussion
+func (s *DiscussionsService) UpdateIssueDiscussion(pid interface{}, issue, discussion int, opt *UpdateIssueDiscussionOptions, options ...OptionFunc) (*Discussion, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/issues/%d/discussions/%d", url.QueryEscape(project), issue, discussion)
+
+	req, err := s.client.NewRequest("PUT", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	n := new(Discussion)
+	resp, err := s.client.Do(req, n)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return n, resp, err
+}
+
+// DeleteIssueDiscussion deletes an existing discussion of an issue.
+//
+// https://docs.gitlab.com/ce/api/discussions.html#delete-an-issue-discussion
+func (s *DiscussionsService) DeleteIssueDiscussion(pid interface{}, issue, discussion int, options ...OptionFunc) (*Response, error) {
```

Comment:
I think this should be called `DeleteIssueDiscussionNote` and it needs an additional `note` parameter that should also be used in the URL. Goes for all delete methods.